src/EditorFeatures/TestUtilities/Remote/InProcRemostHostClient.cs

```diff
@@ -67,12 +70,9 @@ public static async Task<RemoteHostClient> CreateAsync(Workspace workspace, bool
             _inprocServices = inprocServices;
             _remotableDataRpc = remotableDataRpc;
 
-            _rpc = stream.CreateStreamJsonRpc(target: this, inprocServices.Logger);
-
-            // handle disconnected situation
-            _rpc.Disconnected += OnRpcDisconnected;
-
-            _rpc.StartListening();
+            _endPoint = new RemoteEndPoint(stream, inprocServices.Logger, incomingCallTarget: this);
+            _endPoint.Disconnected += OnRpcDisconnected;
```

Comment:
so i find it really confusing trying to wrap my brain around: "this caller should listen for disconnection", and "this caller should listen for disconnection and for unhandled exceptions".

Reply:
If an unexpected exception is thrown during call from VS to ServiceHub, we want to show info bar UI. That's VS specific, so the RemoteEndPoints created on VS side hook that up.

Disconnected - preserved the current behavior where some components wanted to do some action when they loose connection.